### PR TITLE
feat: add snap packaging

### PR DESCRIPTION
**Pull request by tonyespy**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [NA] Tests for the changes have been added (for bug fixes / features)
- [NA] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-cli/blob/master/.github/Contributing.md.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Other information
This PR adds initial snap packaging for edgex-cli. Once published to the store, anyone on a system with snap support will be able to install the client using:

$ sudo snap install edgex-cli

The client can be configured via configuration.toml to run against EdgeX deployed via docker-compose or k8s. The default configuration works with EdgeX deployed natively or via snaps.

### CODE REVIEW

**Comment by tonyespy** [comment]
recheck

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    # changes rarely and therefore will not trigger a new pull
+    source: snap/local
+    override-pull: |
+      cd $SNAPCRAFT_PROJECT_DIR
+      if [ -f VERSION ]; then
+        PROJECT_VERSION=$(cat VERSION)
+      else
+        PROJECT_VERSION=local-dev
+      fi
+      GIT_REVISION=$(git rev-parse --short HEAD)
```
Comment (siggiskulason) [inline snap/snapcraft.yaml:33]: No harm in setting the GIT_REVISION variable, but it's not needed anymore as we we don't use the git revision in the version string.

**Review by siggiskulason — COMMENTED** [review]
The snap builds fine, runs and connects. I also confirmed that the wrapper copies the configuration.toml file to the user config directory as expected and the client picks up the settings from there => LGTM

File: snap/snapcraft.yaml (hunk 1 above)
Comment (tonyespy) [reply snap/snapcraft.yaml]: @siggiskulason Thanks for the review! Just removed the GIT_REVISION line and re-pushed.

**Review by difince — APPROVED** [review]

**Pull request merged**